### ISSUE TITLE
Domain access failed to load during phase: bootstrap include. Please check your settings.php file and site configuration

### ISSUE THREAD

**Issue by sudipto68** [rudy880719/domain#4]
I am getting this error message after installing this module. I have followed the updated Installation.txt file and did the same that are mentioned there but still facing this error every time I refresh the page. Also the domain module's functionality is not working for me.
I have also talked about this error [here](https://github.com/backdrop-ops/contrib/issues/400#issuecomment-2385222849).

**Comment by rudy880719** [comment]
@sudipto68 I couldn't reproduce the error, we could schedule a call to discuss the installation process.

**Comment by sudipto68** [comment]
Sure. Please let me know how can we discuss this.

**Comment by rudy880719** [comment]
@sudipto68 I sent you an invitation on LinkedIn

**Comment by izmeez** [comment]
@sudipto68 Some more details may help. 

1. Can you copy and post the lines you added to your settings.php file? Are they at the end of the file or after the $databases declaration?
2. Can you copy and paste the actual error?

**Comment by sudipto68** [comment]
@izmeez I have talked to rudy and found out the issue why this is happening. I think the difference was in the database configuration and that should be mentioned in the installation.txt file. After the database configuration changes, the module works fine for me. I will test more.

**Comment by izmeez** [comment]
@sudipto68 Can you be more specific or propose what needs to be written up? Possibly share what you changed?

**Comment by rudy880719** [comment]
@izmeez By default, almost all people configure the database parameters in settings.php in this way 

$database = 'mysql://backdrop:backdrop@database/backdrop'

For the domain to work you must set the DB parameters like this:

$databases = array (
  'default' =>
    array (
      'default' =>
        array (
          'database' => 'backdrop',
          'username' => 'backdrop',
          'password' => 'backdrop',
          'host' => 'database',
          'port' => '3306',
          'driver' => 'mysql',
          'prefix' => '',
        ),
    ),
);

**Comment by izmeez** [comment]
Ahh! like the D7 way.
```
$databases['default']['default'] = array(
  'driver' => 'mysql',
  'database' => 'database_name',
  'username' => 'user_name',
  'password' => 'user_password',
  'host' => 'localhost',
  'port' => '',
  'charset' => 'utf8mb4',
  'collation' => 'utf8mb4_unicode_ci',
);
```

**Comment by rudy880719** [comment]
@izmeez yes..!!!!

**Comment by rudy880719** [comment]
@sudipto68 The cache had to be cleared when we edited content, because in the configuration we had the cache activated for anonymous users, if it is disabled, the cache should not be cleared

**Comment by sudipto68** [comment]
@izmeez Today i am testing the module more,  I have done everything that we did yesterday for testing.  But still one domain post is accessible from another domain in incognito. I didn't check the all affiliates and also i have flush the cache. But after the db trick the error has gone.

**Comment by izmeez** [comment]
@sudipto68 Is this a new post or one that was created before?

**Comment by sudipto68** [comment]
@izmeez @rudy880719 I have tested this today again. I don't know if I am missing any rule or not. But it is not wokring properly for me. I have added two posts for two different domain, also added domain source for both of them, at first I didn't check the Send to all affiliates option, then I tried to access the posts from incognito, but i can't access the post, not also in its own domain.
Then I edit one post and checked the Send to all affiliates option, then this post is accessible from both domain in incognito, but this was only for one domain.

* I have cleared the cache every time i edited or added content.
* I rebuild content permission after installing the module.
* I have installed all the domain related module. 

I don't know if I am missing any rule or not.

**Comment by izmeez** [comment]
@sudipto68 Please open a new issue in https://github.com/backdrop-contrib/domain/issues

**Comment by rudy880719** [comment]
@sudipto68 This is exactly how content control works, please consult the documentation for the domain strict module, it describes how content works and its configurations. Remember that if the option send to all affiliates is checked, all anonymous users on all domains will be able to see that content, if the option is not checked, only logged users who belong to the same domain as the content will be able to see it.

**Comment by sudipto68** [comment]
@rudy880719 Yes. But If I made a post for only one domain and I want that anonymous users and logged users will see this post only on this domain, how can I do that?

**Comment by rudy880719** [comment]
@sudipto68 Remember that the option send to all affiliates means that all users of all domains will be able to see it, the Drupal documentation does not specify what you want, if you can try the module in Drupal and tell me if you find what you want.

**Comment by sudipto68** [comment]
@rudy880719 Yes. The domain strict module inforces this rule, for this module's rule the anonymous user can't access the content on it's own domain. I have just disabled the module and test, now anonymous users can access the content too.

**Comment by izmeez** [comment]
@sudipto68 What you have identified is useful. It would be nice if it was summarized in a new issue at the backdrop-contrib/domain repo even as resolved so others can see it. Or if it is added to the wiki at that repo.
